Algorithm.CSharp/ObjectStoreExampleAlgorithm.cs

```diff
@@ -0,0 +1,133 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Data;
+using QuantConnect.Indicators;
+using QuantConnect.Interfaces;
+using QuantConnect.Securities;
+using QuantConnect.Storage;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// This algorithm showcases some features of the <see cref="IObjectStore"/> feature.
+    /// One use case is to make consecutive backtests run faster by caching the results of
+    /// potentially time consuming operations. In this example, we save the results of a
+    /// history call. This pattern can be equally applied to a machine learning model being
+    /// trained and then saving the model weights in the object store.
+    /// </summary>
+    public class ObjectStoreExampleAlgorithm : QCAlgorithm
```

Comment:
It would be nice to have the `py` version, sometimes pythonnet gives surprises 😅